tests/test_upload.py

```diff
@@ -694,6 +694,300 @@ def test_init_object_write_grpc_final_message_empty_data(self):
         self.assertEqual(upload.metadata.name, "object")
         self.assertEqual(upload.metadata.bucket, "projects/_/buckets/bucket-name")
 
+    def test_process_bidi_write_grpc_cannot_resume_completed_upload(self):
```

Comment:
Hmm... why?  Consider this scenario:

- The client is trying to resume an upload
- The client sends a QueryResumableUpload() RPC and gets a result of saying that the upload is not completed.
- Some other client (or a delayed message in the service internal processes) completes the upload
- The client tries to resume the upload by sending the same data as before, and finalizing.

I would think that should succeed with the previously uploaded object.

Reply:
I see this is fixed.  FWIW, don't feel that you need to wait for another review from me.